axelor-human-resource/src/main/java/com/axelor/apps/hr/service/timesheet/TimesheetLineServiceImpl.java

```diff
@@ -216,4 +225,37 @@ public Duration computeTotalDuration(List<TimesheetLine> timesheetLineList) {
     }
     return Duration.ofSeconds(totalSecDuration);
   }
+
+  @Transactional
+  public TimesheetLine setTimesheet(TimesheetLine timesheetLine) {
+    Timesheet timesheet =
+        timesheetRepository
+            .all()
+            .filter(
+                "self.user = ?1 AND self.company = ?2 AND (self.statusSelect = 1 OR self.statusSelect = 2) AND ((?3 BETWEEN self.fromDate AND self.toDate) OR (self.toDate = null)) ORDER BY self.id ASC",
+                timesheetLine.getUser(),
+                timesheetLine.getProject().getCompany(),
+                timesheetLine.getDate())
+            .fetchOne();
+    if (timesheet == null) {
+      timesheet =
+          timesheetRepository
+              .all()
+              .filter(
+                  "self.user = ?1 AND self.statusSelect != ?2 ORDER BY self.toDate DESC",
```

Comment:
Check requirements 
if required add filter on self.company too

Reply:
here no need to filter on company as it is set as user's active company when timesheet created and so it is always one so no need to filter it.